sklearn/metrics/tests/test_classification.py

```diff
@@ -1508,36 +1518,92 @@ def test_precision_recall_f1_score_with_an_empty_prediction():
     assert_almost_equal(f, 1 / 3)
     assert s is None
     assert_almost_equal(fbeta_score(y_true, y_pred, beta=2,
-                                    average="samples"),
+                                    average="samples",
+                                    zero_division=zero_division),
                         0.333, 2)
 
 
 @pytest.mark.parametrize('beta', [1])
 @pytest.mark.parametrize('average', ["macro", "micro", "weighted", "samples"])
-def test_precision_recall_f1_no_labels(beta, average):
+@pytest.mark.parametrize('zero_division', [0, 1])
+def test_precision_recall_f1_no_labels(beta, average, zero_division):
     y_true = np.zeros((20, 3))
     y_pred = np.zeros_like(y_true)
 
-    p, r, f, s = assert_warns(UndefinedMetricWarning,
-                              precision_recall_fscore_support,
-                              y_true, y_pred, average=average,
-                              beta=beta)
+    p, r, f, s = assert_no_warnings(precision_recall_fscore_support, y_true,
+                                    y_pred, average=average, beta=beta,
+                                    zero_division=zero_division)
+    fbeta = assert_no_warnings(fbeta_score, y_true, y_pred, beta=beta,
+                               average=average, zero_division=zero_division)
+
+    zero_division = float(zero_division)
+    assert_almost_equal(p, zero_division)
+    assert_almost_equal(r, zero_division)
+    assert_almost_equal(f, zero_division)
+    assert s is None
+
+    assert_almost_equal(fbeta, float(zero_division))
+
+
+@pytest.mark.parametrize('average', ["macro", "micro", "weighted", "samples"])
+def test_precision_recall_f1_no_labels_check_warnings(average):
+    y_true = np.zeros((20, 3))
+    y_pred = np.zeros_like(y_true)
+
+    func = precision_recall_fscore_support
+    p, r, f, s = assert_warns(UndefinedMetricWarning, func, y_true, y_pred,
```

Comment:
Can we change the assert_warns into `pytest.warns` context managers?

```py
with pytest.warns(UndefinedMetricWarning):
	p, r, f, s = precision_recall_fscore_support(y_true, y_pred, ...)
```

and the same for the other warning assertions.

Reply:
done